workflow/controller/controller.go

```diff
@@ -584,8 +604,50 @@ func (wfc *WorkflowController) getPod(namespace string, podName string) (*apiv1.
 	return pod, nil
 }
 
+func (wfc *WorkflowController) enablePodForDeletion(ctx context.Context, pods typedv1.PodInterface, namespace string, podName string, extraPatches ...PatchOperation) error {
+	var patches []PatchOperation
+	pod, err := wfc.getPodFromAPI(ctx, namespace, podName)
+	if err != nil {
+		return err
+	}
+	patch := createFinalizerRemovalPatchIfExists(pod, common.FinalizerPodStatus)
+	if patch != nil {
+		patches = append(patches, *patch)
+	}
+	patches = append(patches, extraPatches...)
+	if err := applyPatches(ctx, pods, pod.Name, patches); err != nil {
+		return err
+	}
+	return nil
+}
+
+func createFinalizerRemovalPatchIfExists(pod *apiv1.Pod, targetFinalizer string) *PatchOperation {
+	i := slices.Index(pod.Finalizers, targetFinalizer)
+	if i >= 0 {
+		return &PatchOperation{
+			Operation: "remove",
+			Path:      fmt.Sprintf("/metadata/finalizers/%d", i),
+		}
+	}
+	return nil
+}
+
+func applyPatches(ctx context.Context, pods typedv1.PodInterface, podName string, patches []PatchOperation) error {
+	if len(patches) == 0 {
+		log.Debug("not patching pod")
```

Comment:
will this log line include information about which pod it refers to?

Reply:
same comment for line below

Reply:
This format follows the original PR. Do I need to make any changes?

Reply:
can you add the Pod name to it?

Reply:
I've added in the bellow commits.
https://github.com/argoproj/argo-workflows/pull/12413/commits/7d2640d03539df1952e74f3e8afe502980276d66

This format followed
https://github.com/argoproj/argo-workflows/blob/a15587755e250fbbe3b9538b1c15ae581a108f0c/workflow/controller/exec_control.go#L45

When I check the logs, it looks good for me.
```
controller: time="2024-01-26T08:13:32.358Z" level=debug msg="not patching pod" podName=test-pod-cleanup-on-pod-completion-w6f9d
```

```
controller: time="2024-01-26T08:11:24.619Z" level=debug msg="patching pod" data="[{\"op\":\"remove\",\"path\":\"/metadata/finalizers/0\"}]" podName=test-pod-cleanup-on-pod-completion-label-selected-2dxbn
```

Reply:
great, thanks, yeah, I wasn't sure what was already in the `log` - thanks for checking!